barretenberg/cpp/src/barretenberg/relations/poseidon2_internal_relation.hpp

```diff
@@ -62,53 +62,61 @@ template <typename FF_> class Poseidon2InternalRelationImpl {
     {
         PROFILE_THIS_NAME("PoseidonInt::accumulate");
         using Accumulator = std::tuple_element_t<0, ContainerOverSubrelations>;
-        using View = typename Accumulator::View;
-        auto w_l = View(in.w_l);
-        auto w_r = View(in.w_r);
-        auto w_o = View(in.w_o);
-        auto w_4 = View(in.w_4);
-        auto w_l_shift = View(in.w_l_shift);
-        auto w_r_shift = View(in.w_r_shift);
-        auto w_o_shift = View(in.w_o_shift);
-        auto w_4_shift = View(in.w_4_shift);
-        auto q_l = View(in.q_l);
-        auto q_poseidon2_internal = View(in.q_poseidon2_internal);
+        using MonomialAccumulator = typename Accumulator::MonomialAccumulator;
+
+        auto w_l_m = MonomialAccumulator(in.w_l);
+        auto w_l_shift_m = MonomialAccumulator(in.w_l_shift);
+        auto w_r_shift_m = MonomialAccumulator(in.w_r_shift);
+        auto w_o_shift_m = MonomialAccumulator(in.w_o_shift);
+        auto w_4_shift_m = MonomialAccumulator(in.w_4_shift);
+        auto q_l_m = MonomialAccumulator(in.q_l);
+        auto q_poseidon2_internal_m = MonomialAccumulator(in.q_poseidon2_internal);
 
         // add round constants
-        auto s1 = w_l + q_l;
+        auto s1 = Accumulator(w_l_m + q_l_m);
 
         // apply s-box round
         auto u1 = s1.sqr();
         u1 = u1.sqr();
         u1 *= s1;
-        auto u2 = w_r;
-        auto u3 = w_o;
-        auto u4 = w_4;
+        auto u2_m = MonomialAccumulator(in.w_r);
+        auto u3_m = MonomialAccumulator(in.w_o);
+        auto u4_m = MonomialAccumulator(in.w_4);
 
+        auto q_pos_by_scaling_m = (q_poseidon2_internal_m * scaling_factor);
+        auto q_pos_by_scaling = Accumulator(q_pos_by_scaling_m);
```

Comment:
The call to `Accumulator` is because we need extra values to properly perform subsequent computations right?

Reply:
correct